redbot/cogs/mod/slowmode.py

```diff
@@ -0,0 +1,44 @@
+import re
+from .abc import MixinMeta
+from datetime import timedelta
+from redbot.core import commands, i18n, checks
+from redbot.core.utils.chat_formatting import humanize_timedelta
+
+_ = i18n.Translator("Mod", __file__)
+
+
+class Slowmode(MixinMeta):
+    """
+    Commands regarding channel slowmode management.
+    """
+
+    @commands.command()
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_channels=True)
+    @checks.admin_or_permissions(manage_channels=True)
+    async def slowmode(self, ctx, interval: str = None):
+        """Changes channel's slowmode setting.
+
+        Interval can be anything from 0 seconds to 6 hours.
+        You can specify the unit using `s`, `m` or `h` as suffix.
+        Use without parameters or set to 0 to disable.
+        """
+        if interval:
+            match = re.match(r"(?i)(\d{1,5})([a-zA-Z])?$", interval)
+            if match:
+                interval = int(match.group(1))
+                if match.group(2) is not None:
+                    interval = {"s": 1, "m": 60, "h": 60 * 60}.get(
+                        match.group(2).lower(), -1
+                    ) * interval
+            if not match or not 0 <= interval <= 21600:

```

Comment:
These negative conditions should be handled differently.

To a user, getting this feedback as a result of this input isn't clear what they did wrong. Sending the help on a non match is fine here.

![image](https://user-images.githubusercontent.com/22035876/58454824-ac6deb00-80ed-11e9-95a1-fdb797e20cfd.png)

Reply:
This should be resolved once #2736 will be accepted, so I can use it here.